models/packages/package.go

```diff
@@ -120,25 +120,30 @@ type Package struct {
 
 // TryInsertPackage inserts a package. If a package exists already, ErrDuplicatePackage is returned
 func TryInsertPackage(ctx context.Context, p *Package) (*Package, error) {
-	e := db.GetEngine(ctx)
+	n, err := db.InsertOnConflictDoNothing(ctx, p)
+	if err != nil {
+		return nil, err
+	}
+	if n != 0 {
+		return p, nil
```

Comment:
The insert methods need to return the filled bean. For example it's expected that `ID` has a value in the return value which is not the case here.

Reply:
The code auto fills the ID of the bean.